router/pkg/config/config.go

```diff
@@ -334,13 +334,28 @@ type EngineExecutionConfiguration struct {
 	EnableSubgraphFetchOperationName       bool                     `envDefault:"false" env:"ENGINE_ENABLE_SUBGRAPH_FETCH_OPERATION_NAME" yaml:"enable_subgraph_fetch_operation_name"`
 }
 
+type BlockMutationConfiguration struct {
+	Enabled   bool   `yaml:"enabled" envDefault:"false" env:"BLOCK_MUTATION_ENABLED"`
+	Condition string `yaml:"condition" env:"BLOCK_MUTATION_CONDITION"`
+}
+
+type BlockSubscriptionConfiguration struct {
+	Enabled   bool   `yaml:"enabled" envDefault:"false" env:"BLOCK_SUBSCRIPTION_ENABLED"`
+	Condition string `yaml:"condition" env:"BLOCK_SUBSCRIPTION_CONDITION"`
+}
+
+type BlockNonPersistedConfiguration struct {
+	Enabled   bool   `yaml:"enabled" envDefault:"false" env:"BLOCK_NON_PERSISTED_ENABLED"`
+	Condition string `yaml:"condition" env:"BLOCK_NON_PERSISTED_CONDITION"`
+}
+
```

Comment:
These types basically are the same and could be simplified into a single type

```suggestion
type BlockConfiguration struct {
	Enabled   bool   `yaml:"enabled" envDefault:"false" env:"ENABLED"`
	Condition string `yaml:"condition" env:"CONDITION"`
}
```

The SecurityConfiguration type could then be replaced with `envPrefix`

Also I think in general the 3 fields cannot have an `env` config anyways as they are complex types?

```go
type SecurityConfiguration struct {
	BlockMutations              BlockMutationConfiguration     `yaml:"block_mutations" envPrefix:"SECURITY_BLOCK_MUTATIONS_"`
	BlockSubscriptions          BlockSubscriptionConfiguration `yaml:"block_subscriptions" envPrefix:"SECURITY_BLOCK_SUBSCRIPTIONS_"`
	BlockNonPersistedOperations BlockNonPersistedConfiguration `yaml:"block_non_persisted_operations" envPrefix:"SECURITY_BLOCK_NON_PERSISTED_OPERATIONS_"`
```

Reply:
I like. I use envPrefix.